scan/src/utils/Route.re

```diff
@@ -39,87 +39,89 @@ type t =
-  | (["oracle-script", oracleScriptID], "revisions") =>
-    OracleScriptIndexPage(oracleScriptID |> int_of_string, OracleScriptRevisions)
-  | (["oracle-script", oracleScriptID], _) =>
-    OracleScriptIndexPage(oracleScriptID |> int_of_string, OracleScriptExecute)
+  | (["sources"], _) => DataSourceHomePage
+  | (["scripts"], _) => OracleScriptHomePage
   | (["txs"], _) => TxHomePage
-  | (["tx", txHash], _) => TxIndexPage(Hash.fromHex(txHash))
   | (["validators"], _) => ValidatorHomePage
   | (["blocks"], _) => BlockHomePage
-  | (["block", blockHeight], _) =>
-    let blockHeightIntOpt = blockHeight |> int_of_string_opt;
-    BlockIndexPage(blockHeightIntOpt->Belt_Option.getWithDefault(0));
-  | (["request", reqID], "proof") => RequestIndexPage(reqID |> int_of_string, RequestProof)
-  | (["request", reqID], _) => RequestIndexPage(reqID |> int_of_string, RequestReportStatus)
-  | (["account", address], "delegations") =>
+  | (["tx", txHash], _) => TxIndexPage(Hash.fromHex(txHash))
+  | ([address], "delegations") =>
     AccountIndexPage(address |> Address.fromBech32, AccountDelegations)
-  | (["account", address], _) =>
-    AccountIndexPage(address |> Address.fromBech32, AccountTransactions)
-  | (["validator", address], "delegators") =>
-    ValidatorIndexPage(address |> Address.fromBech32, Delegators)
-  | (["validator", address], "reports") =>
-    ValidatorIndexPage(address |> Address.fromBech32, Reports)
-  | (["validator", address], _) =>
-    ValidatorIndexPage(address |> Address.fromBech32, ProposedBlocks)
+  | ([address], "delegators") => ValidatorIndexPage(address |> Address.fromBech32, Delegators)
+  | ([address], "reports") => ValidatorIndexPage(address |> Address.fromBech32, Reports)
+  | ([path], tab) =>
+    switch ((path |> Js.String.split(""))->Belt_List.fromArray, tab) {
+    | (["B", ...rest], _) => BlockIndexPage(rest |> chars_to_int)
+    | (["O", ...rest], "code") => OracleScriptIndexPage(rest |> chars_to_int, OracleScriptCode)
+    | (["O", ...rest], "requests") =>
+      OracleScriptIndexPage(rest |> chars_to_int, OracleScriptRequests)
+    | (["O", ...rest], "revisions") =>
+      OracleScriptIndexPage(rest |> chars_to_int, OracleScriptRevisions)
+    | (["O", ...rest], _) => OracleScriptIndexPage(rest |> chars_to_int, OracleScriptExecute)
+    | (["D", ...rest], "code") => DataSourceIndexPage(rest |> chars_to_int, DataSourceCode)
+    | (["D", ...rest], "requests") =>
+      DataSourceIndexPage(rest |> chars_to_int, DataSourceRequests)
+    | (["D", ...rest], "revisions") =>
+      DataSourceIndexPage(rest |> chars_to_int, DataSourceRevisions)
+    | (["D", ...rest], _) => DataSourceIndexPage(rest |> chars_to_int, DataSourceExecute)
+    | (["R", ...rest], "proof") => RequestIndexPage(rest |> chars_to_int, RequestProof)
+    | (["R", ...rest], _) => RequestIndexPage(rest |> chars_to_int, RequestReportStatus)
+    | (["b", "a", "n", "d", "v", "a", "l", "o", "p", "e", "r", ..._], _) =>
```

Comment:
Seem weird use startwith like search will be better?  @sorawit

Reply:
My recommendation is that we should explore using regex matching / group extraction. @prin-r already started looking into this yesterday.